public/app/features/templating/state/applyStateChanges.ts

```diff
@@ -0,0 +1,7 @@
+export type MutateStateFunc<S> = (state: S) => S;
+
+export const applyStateChanges = <S>(state: S, ...args: Array<MutateStateFunc<S>>): S => {
```

Comment:
Maybe this should be moved to app/core/utils ?

Reply:
fixed!